pkg/controllers/customcluster_helper.go

```diff
@@ -413,3 +415,126 @@ func hasProvisionClusterInfo(phase v1alpha1.CustomClusterPhase) bool {
+	if err != nil {
+		return err
+	}
+	defer sftpClient.Close()
+
+	kubeConfigData, err := r.fetchRemoteKubeConfig(sftpClient, getProvisionedKubeConfigPath())
+	if err != nil {
+		return err
+	}
+
+	err = r.createKubeConfigSecret(ctx, generateProvisionedClusterSecretName(customCluster), customCluster.Namespace, kubeConfigData)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *CustomClusterController) getSSHKeySecret(ctx context.Context, namespace, name string) (*corev1.Secret, error) {
+	sshKeySecret := &corev1.Secret{}
+	key := client.ObjectKey{Namespace: namespace, Name: name}
+	if err := r.Client.Get(ctx, key, sshKeySecret); err != nil {
+		return nil, err
+	}
+	return sshKeySecret, nil
+}
+
+func (r *CustomClusterController) buildSSHClientConfig(sshKeySecret *corev1.Secret) (*ssh.ClientConfig, error) {
+	sshPrivateKeyData, ok := sshKeySecret.Data["ssh-privatekey"]
+	if !ok {
+		return nil, fmt.Errorf("ssh-privatekey not found in secret")
+	}
+
+	signer, err := ssh.ParsePrivateKey(sshPrivateKeyData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse SSH private key: %v", err)
+	}
+
+	sshConfig := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	return sshConfig, nil
+}
+
+func (r *CustomClusterController) connectSFTPClient(addr string, sshConfig *ssh.ClientConfig) (*sftp.Client, error) {
+	conn, err := ssh.Dial("tcp", addr, sshConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect: %v", err)
+	}
+
+	sftpClient, err := sftp.NewClient(conn)
+	if err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("failed to create SFTP client: %v", err)
+	}
+
+	return sftpClient, nil
+}
+
+func (r *CustomClusterController) fetchRemoteKubeConfig(sftpClient *sftp.Client, path string) ([]byte, error) {
+	remoteFile, err := sftpClient.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open remote file: %v", err)
+	}
+	defer remoteFile.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(remoteFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read remote file: %v", err)
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (r *CustomClusterController) createKubeConfigSecret(ctx context.Context, name, namespace string, kubeConfigData []byte) error {
+	newSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"admin.conf": kubeConfigData,
+		},
+	}
+
+	if err := r.Client.Create(ctx, newSecret); err != nil {
+		return fmt.Errorf("failed to create new secret: %v", err)
+	}
+
+	return nil
+}
+
+func getProvisionedKubeConfigPath() string {
+	return "/etc/kubernetes/admin.conf"
+}
+
+func generateProvisionedClusterSecretName(customCluster *v1alpha1.CustomCluster) string {
+	return provisionedClusterKubeConfigSecretPrefix + customCluster.Name
```

Comment:
THis is too long, it may exceed max allowed length

Reply:
Why do you mark it resolved

Reply:
I had change 

func generateProvisionedClusterSecretName(customCluster *v1alpha1.CustomCluster) string {
	return provisionedClusterKubeConfigSecretPrefix + customCluster.Name" 

to 

func getKubeConfigSecretName(customCluster *v1alpha1.CustomCluster) string {
	return provisionedKubeConfigPrefix + customCluster.Name

in https://github.com/kurator-dev/kurator/pull/279/commits/ad5e12994285be1a6b36ce2407180dd5f2ce02e5

if this is still too lang

Reply:
why do you need prefix